crates/floresta-chain/src/pruned_utreexo/mod.rs

```diff
@@ -351,23 +355,116 @@ impl<T: BlockchainInterface> BlockchainInterface for Arc<T> {
     }
 }
+
+pub mod utxo_data {
+    use super::*;
+
+    pub struct UtxoData {
+        /// The transaction output that created this UTXO.
+        pub txout: bitcoin::TxOut,
+        //we need this mostly to validate relative-locktime.
+        /// The height of the block that the utxo was commited.
+        pub height: u32,
+        /// The timestamp of the block that the utxo was commited.
+        pub time: u32,
+    }
+
+    impl UtxoData {
+        /// Creates a new UtxoData.
+        pub const fn new(txout: bitcoin::TxOut, height: u32, time: u32) -> Self {
+            Self {
+                txout,
+                height,
+                time,
+            }
+        }
+        /// Gets the output.
+        pub const fn get_txout(&self) -> &bitcoin::TxOut {
+            &self.txout
+        }
+        /// Gets the height that the output was commited.
+        pub const fn get_height(&self) -> u32 {
+            self.height
+        }
+        pub const fn get_time(&self) -> u32 {
+            self.time
+        }
+    }
+    /// A [`HashMap<bitcoin::OutPoint, UtxoData>`] alias.
+    ///
+    /// Use HashMap to store and fast-find all needed UTXOs related data.
+    pub type UtxoMap = HashMap<bitcoin::OutPoint, UtxoData>;
+}
+
+pub mod chain_time {
+    /// This module contain only a trait to delegate local-time and Median Time Past.
+    /// 
+    /// The consumer of `Floresta-chain` has the option to implement it or just pass any (chaintime, mtp) as (u32,u32)
+    /// 
+    /// # Examples
+    /// ```
+    /// 
+    /// fn check_chain_time<C: Chaintime>(chain_time_obj: C) -> bool {
+    ///     if !(let u32 = chain_time_obj.get_time()){
+    ///         return false;
+    ///     }
+    ///     if !(let u32 = chain_time_obj.get_mtp()){
+    ///         return false;
+    ///     }
+    ///     true 
+    /// }
+    /// 
+    /// //(u32,u32) already have a impl inside chain_time module.
+    /// 
+    /// let zero_unixtimestamp: u32 = 0; // zero values will disable time checks
+    /// 
+    /// let zero_mtp: u32 = 0; // zero values will disable time checks
+    /// 
+    /// assert(check_chain_time((direct_zero_in,zero_mtp)))
+    /// 
+    /// ```
+    pub type MinimalChainTime = (u32,u32);
+
+    pub const DISABLE_TIME:MinimalChainTime = (0,0);
+        pub trait Chaintime {
+
+        /// get_time -> UnixTimestamp
+        ///
+        /// Should return a unix timestamp or 0 when to skip any time related validation.    
+        fn get_time(&self) -> u32;
+
+        /// get_mtp -> BlockTimestamp
+        ///
+        /// Should return a BlockTimestamp or 0 when to skip any mtp related validation.   
+        fn get_mtp(&self) -> u32;
+    }
+
+    //Heres a little impl so we can insert any u32. 
+    impl Chaintime for MinimalChainTime {
+        fn get_time(&self) -> u32 {
+            self.0.clone()
+        }
+        fn get_mtp(&self) -> u32 {
+            self.1.clone()
+        }
+    }
+}
```

Comment:
Now is a trait and we dont loose control and functionality because of the shipped tuple that can be used as example.